### DISCUSSION TITLE
Keycloak as Security Token Service (STS) for SAML2

### DISCUSSION THREAD

**Discussion by jochenr** [keycloak/keycloak#24156] [Q&A]
Hi all,

today I spent a lot of time in searching for documentaion how Keycloak can be used as ecurity Token Service (STS) for SAML2 (Holder-of-Key).

I thought that Keycloak supports this, because I can create a SAML client. But after creating one, it seems there is no entpoint for the WS-Trust STS service.

Searching the internet the best articles/projects I found are:

https://github.com/berkeragir/cxf-sts-keycloak-demo/tree/master
and
https://coheigea.blogspot.com/2018/06/combining-keycloak-with-apache-cxf-sts.html

These readings confuse me, because my expecation was that Keycloak can work as STS itself?!?

Is there a way to do SAML2 Holder-of-Key directly with Keycloak?

Thanks

**Comment by deepakgangore** [comment]
hi @jochenr , I see this query was posted last year. Have you got answer to your query? I am also looking for this solution. Could you please share if you have got answer.

Thanks

**Reply by jochenr** [reply]
Hi @deepakgangore,

unfortunetely there is no plan, including this feature😕

We still don‘t have a solution, but I don't have the time to implement (and maintain) it myself.

Perhaps you can raisea new feature  request. I'll Vote for it.

Best regards,
Jochen
